pkg/apis/core/validation/validation.go

```diff
@@ -3024,6 +3025,44 @@ func ValidatePod(pod *core.Pod) field.ErrorList {
 		allErrs = append(allErrs, field.Invalid(specPath, hugePageResources, "must use a single hugepage size in a pod spec"))
 	}
 
+	podIPsField := field.NewPath("status", "podIPs")
+
+	// all PodIPs must be valid IPs
+	for idx, podIP := range pod.Status.PodIPs {
+		if net.ParseIP(podIP.IP) == nil {
+			allErrs = append(allErrs, field.Invalid(podIPsField, pod.Status.PodIPs, fmt.Sprintf("ip %v at index(%v) is not a valid PodIP", podIP.IP, idx)))
+		}
+	}
+
+	// dualstack feature must be enabled if we have more than one PodIP
+	if len(pod.Status.PodIPs) > 1 && !utilfeature.DefaultFeatureGate.Enabled(features.IPv6DualStack) {
+		allErrs = append(allErrs, field.Invalid(podIPsField, pod.Status.PodIPs, "multiple PodIPs were found and dual stack feature is not enabled"))
+	}
+
+	// if we have more than one Pod.PodIPs then we validate for dual stack
+	if len(pod.Status.PodIPs) > 1 {
+		podIPs := make([]string, 0, len(pod.Status.PodIPs))
+		for _, podIP := range pod.Status.PodIPs {
+			podIPs = append(podIPs, podIP.IP)
+		}
+		dualStack, err := netutils.IsDualStackIPStrings(podIPs)
+		if err != nil {
+			allErrs = append(allErrs, field.Invalid(podIPsField, pod.Status.PodIPs, fmt.Sprintf("Failed to check with dual stack with error:%v", err)))
+		}
+		if !dualStack {
+			allErrs = append(allErrs, field.Invalid(podIPsField, pod.Status.PodIPs, "expected to be dual stack (at least an ip from each ipfamily)"))
```

Comment:
"may have at most one value for each address family (IPv4, IPv6)"

Reply:
we removed `len(...)==2` check, so a better error message is `at least..`